### DISCUSSION TITLE
Mac version

### DISCUSSION THREAD

**Discussion by abulka** [aurelienpierreeng/ansel#58] [Q&A]
I agree with everything written about the need for this fork, only to find a build for Mac is missing. 

Darktable for Mac has always been a second class citizen esp. re performance. Its pretty bad. I'd love to get a feel for how the Ansell changes affect performance on a Mac - both x86 and arm. Though I understand that until proper use is made of Apple Metal graphics, there wil always be limits to performace on a Mac.

I built Darktable once on my M1 Macmini but not keen on going through that process again because, whilst it built and ran, there were some database/cache/config issues that were out of my depth, given that I am pretty new to Darktable.

Can anyone help re building Ansel for Mac and sharing their binaries? Or even just some clear build instructions would be great. The generic darktable build instructions don't even work for Darktable anymore - see [this issue](https://github.com/darktable-org/darktable/issues/13314). Even so, Ansel lacks the handy build scripts from Darktable see https://github.com/darktable-org/darktable/tree/master/packaging/macosx - not sure why they weren't forked as, on a good day, perhaps they might have worked to compile Ansel.

**Comment by aurelienpierre** [comment]
I will try again, _at some point_, to ship a pre-built Mac package, but it looks grim. 

For the record, I fixed the Mac CI bot 3 times in december 2022, and it kept breaking again every week. The culprit is how Mac Brew is unable to manage conflicting dependencies (in this case, Python version) and regularly breaks, so the CI broke at dependency install.

So the last stroke made me disable Mac CI at all. Mac users weigh for less than 9% of the darktable user base, it's unfair to the 91 % of people using another OS to divert that much ressources into fixing third-party mess.

**Comment by aurelienpierre** [comment]
Something I need to stress, is that Mac is not a second-class citizen neither for Ansel nor darktable. The support of open-source libraries on Mac is a third-class citizen and Apple want us to rewrite everything with their own techs and API, which is never going to happen. We can ship software to Linux and Windows with little changes, but I think users don't nearly understand what it takes to properly support Mac, given that half of the dependencies of dt/Ansel are semi-broken on this OS.

**Comment by rfeg** [comment]
Hi Aurelien,

First thanks for the good work. 

I understand that mac is not a first option for open source sw, but for many of us (users that want safety and NOT babysit or take good care of our tools is a good option) and now my M1 macbook is the only true silent photo editing PC option that I know of.

Instead of deviating your time to 9% of user base, can you explain why the macports git is only intel only? 
    (https://github.com/aurelienpierreeng/ansel/blob/master/packaging/macosx/BUILD.txt)
What is needed to be also arm compatible?

Thanks in advance,
RG

PS - I also find annoyances on dt, hoping that ansel will solve some; my setup is digicam + dt (filmic fan) on a macbook M1, so I'm probably 1%

**Comment by rfeg** [comment]
Hi Aurelien,
I think I found an away to please a big part of 9% share of mac os users, at least to help you test and validate your fork: can you compile to arm64 ubuntu? 

I managed to install ubuntu on M1 mac so if you use your make/compile but for arm64 and make available in the website you can get my and others' help.

(you may argue that we can build from source, but that is not my expertise.

Thank you very much.

Regards,
rfeg
